### ISSUE TITLE
Insert at specific index/key in nn.Sequential

### ISSUE THREAD

**Issue by FrancescoSaverioZuppichini** [pytorch/pytorch#43876]
## 🚀 Feature
It would be great if we can insert a module in a specific position inside a `nn.Sequential`. This allows for more customization. For example. Assuming somebody provides me this `block`

```python
block = nn.Sequential(OrderedDict({ 
    'conv1' : nn.Conv2d(32, 64, kernel_size=3),
    'act1': nn.ReLU(),
    'conv2': nn.Conv2d(64, 128, kernel_size=3),
    'act2': nn.ReLU(),

}))
```python
Maybe I want to add batch norm. 

```
#proposed api
block.insert(name='bn', module=nn.Batchnorm2d(64), after='conv1')
```

## Motivation

Usually, in CV, we want to change already made components, by directly insert a module inside another we can write less code and have better customization.


Stay safe!

Cheers,

Francesco

**Comment by mariosasko** [comment]
I would advise you to store modules inside `nn.ModuleList`, which supports insertion, and layer names in a separate dict/list. Then you can easily insert new modules at a specific position and iterate over them in the forward pass.

**Comment by FrancescoSaverioZuppichini** [comment]
Hi @mariosasko. It is not the same thing and the scope is different. First, you have to write a `forward` and you cannot use it directly. Second, imagine that you want to extend a model written by another person. Been able to insert modules in a `nn.Sequential` should be the way to go in my opinion. It allows more extensibility and I cannot see why this can hurt :)

**Comment by mruberry** [comment]
Thank you for the feature suggestion, @FrancescoSaverioZuppichini. I think you can implement this functionality for yourself using a little Python even if PyTorch doesn't offer it. A Sequential object is a Module, and the Sequential constructor inserts the given Modules into the Module's _module OrderedDict. So I think you could acquire this OrderedDict from the object and rearrange the modules in it (or replace it)?

**Comment by FrancescoSaverioZuppichini** [comment]
Hi, @mruberry. Thank you for your reply,  yes I was thinking about something like that but I think could be a nice addition to the PyTorch ecosystem.

**Comment by vadimkantorov** [comment]
This would also be useful for the case where calling the `super().__init__()` already constructs the sequential and some inserts are needed, e.g. quant/dequant modules: https://github.com/pytorch/vision/issues/3331
